crd-convert/src/main/java/io/strimzi/kafka/crd/convert/cli/JYCommand.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.crd.convert.cli;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import io.strimzi.kafka.crd.convert.utils.ContentType;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+
+/**
+ * Json or Yaml content.
+ */
+public abstract class JYCommand extends ContentCommand {
+
+    @CommandLine.Option(names = {"-t", "--type"}, description = "Content type, e.g. json or yaml")
+    ContentType type;
+
+    protected abstract JsonNode run(JsonNode root) throws IOException;
+
+    @Override
+    protected Object run(byte[] data) throws IOException {
+        ContentType ct = type;
+        if (ct == null && extension != null) {
+            ct = ContentType.findByExtension(extension);
+        }
+        if (ct == null) {
+            throw new IllegalStateException("Cannot determine content type!");
+        }
+        JsonMapper mapper = ct.getMapper();
+        JsonNode result = run(mapper.readTree(data));
```

Comment:
Just an idea, but rather than messing around with extensions I would just parse it as YAML. That should work because YAML is a superset of JSON, IIRC. It also makes it easier to have the command operate on stdin, which would make it a little more flexible.

Reply:
So always yaml, and the parser will know from the content?
The problem here, afaik, it's the output, as it will then always be yaml (or perhaps always json).

Reply:
Well you could support an `--output-format` option for that. My feeling is people would most commonly be using YAML, in which case defaulting to YAML for output would be fine, and would enable reading from stdin (and thus in a shell pipeline). Having an option to change the output to JSON if people really wanted wouldn't be difficult to implement. Combine both approaches and people would only need to use such an option when reading from stdin and wanting JSON output, which would be relatively uncommon, I think.